contracts/utils/cryptography/MultiSignerERC7913.sol

```diff
@@ -0,0 +1,225 @@
+            require(_signersSet.add(signer), MultiSignerERC7913AlreadyExists(signer));
+        }
+        emit ERC7913SignersAdded(newSigners);
+    }
+
+    /// @dev Removes the `oldSigners` from the authorized signers. Internal version without access control.
+    function _removeSigners(bytes[] memory oldSigners) internal virtual {
+        for (uint256 i = 0; i < oldSigners.length; i++) {
+            bytes memory signer = oldSigners[i];
+            require(_signersSet.remove(signer), MultiSignerERC7913NonexistentSigner(signer));
+        }
+        _validateReachableThreshold();
+        emit ERC7913SignersRemoved(oldSigners);
+    }
+
+    /// @dev Sets the signatures `threshold` required to approve a multisignature operation. Internal version without access control.
+    function _setThreshold(uint256 newThreshold) internal virtual {
+        _threshold = newThreshold.toUint128();
+        _validateReachableThreshold();
+        emit ERC7913ThresholdSet(newThreshold);
+    }
+
+    /// @dev Validates the current threshold is reachable.
+    function _validateReachableThreshold() internal view virtual {
+        uint256 totalSigners = _signers().length();
+        uint256 currentThreshold = threshold();
+        require(
+            totalSigners >= currentThreshold,
+            MultiSignerERC7913UnreachableThreshold(totalSigners, currentThreshold)
+        );
+    }
+
+    /**
+     * @dev Decodes, validates the signature and checks the signers are authorized.
+     * See {_validateNSignatures} and {_validateThreshold} for more details.
+     *
+     * Example of signature encoding:
+     *
+     * ```solidity
+     * // Encode signers (verifier || key)
+     * bytes memory signer1 = abi.encodePacked(verifier1, key1);
+     * bytes memory signer2 = abi.encodePacked(verifier2, key2);
+     *
+     * // Order signers by their id
+     * if (keccak256(signer1) > keccak256(signer2)) {
+     *     (signer1, signer2) = (signer2, signer1);
+     *     (signature1, signature2) = (signature2, signature1);
+     * }
+     *
+     * // Assign ordered signers and signatures
+     * bytes[] memory signers = new bytes[](2);
+     * bytes[] memory signatures = new bytes[](2);
+     * signers[0] = signer1;
+     * signatures[0] = signature1;
+     * signers[1] = signer2;
+     * signatures[1] = signature2;
+     *
+     * // Encode the multi signature
+     * bytes memory signature = abi.encode(signers, signatures);
+     * ```
+     */
+    function _rawSignatureValidation(
+        bytes32 hash,
+        bytes calldata signature
+    ) internal view virtual override returns (bool) {
+        if (signature.length == 0) return false; // For ERC-7739 compatibility
+        (bytes[] memory signingSigners, bytes[] memory signatures) = abi.decode(signature, (bytes[], bytes[]));
+        if (signingSigners.length != signatures.length) return false;
+        return _validateNSignatures(hash, signingSigners, signatures) && _validateThreshold(signingSigners);
```

Comment:
I would call `_validateThreshold` first to short circuit failure if that is invalid

Reply:
Addressed in https://github.com/OpenZeppelin/openzeppelin-community-contracts/pull/125/commits/e5bfa3e9905bf0c68b2d51b165a66654b36c28c5